package com.yrgo.dataaccess;

import com.yrgo.domain.Call;
import com.yrgo.domain.Customer;
import org.springframework.jdbc.core.JdbcTemplate;

import java.util.List;

public class CustomerDaoJdbcTemplateImpl implements CustomerDao {
    private static final String CREATE_CUSTOMER_TABLE = "CREATE TABLE " + "CUSTOMER " +
            "(CUSTOMER_ID integer generated by default as identity (start with 1), " +
            "COMPANY_NAME VARCHAR(255), " +
            "EMAIL VARCHAR(255), " +
            "TELEPHONE VARCHAR(30), " +
            "NOTES VARCHAR(255))";
    private static final String CREATE_CUSTOMER_CALL_TABLE = "CREATE TABLE " + "CUSTOMER_CALLS " +
            "(CALL_ID integer generated by default as identity (start with 1), " +
            "TIME_AND_DATE DATE," +
            "NOTES VARCHAR(255), " +
            "CUSTOMER_ID VARCHAR(20))";

    private JdbcTemplate template;

    CustomerDaoJdbcTemplateImpl(JdbcTemplate template) {
        this.template = template;
    }

    @Override
    public void create(Customer customer) {

    }

    private void createTables()	{
        try{
            this.template.update(CREATE_CUSTOMER_TABLE);
            this.template.update(CREATE_CUSTOMER_CALL_TABLE);
        }catch (org.springframework.jdbc.BadSqlGrammarException e){
            System.out.println("Assuming the Action table exists");
        }
    }

    @Override
    public Customer getById(String customerId) throws RecordNotFoundException {
        return null;
    }

    @Override
    public List<Customer> getByName(String name) {
        return List.of();
    }

    @Override
    public void update(Customer customerToUpdate) throws RecordNotFoundException {

    }

    @Override
    public void delete(Customer oldCustomer) throws RecordNotFoundException {

    }

    @Override
    public List<Customer> getAllCustomers() {
        return List.of();
    }

    @Override
    public Customer getFullCustomerDetail(String customerId) throws RecordNotFoundException {
        return null;
    }

    @Override
    public void addCall(Call newCall, String customerId) throws RecordNotFoundException {

    }
}
